signingscript/sign.py

```diff
@@ -625,3 +642,59 @@ def _get_tarfile_compression(compression):
         return to
     except Exception as e:
         raise SigningScriptError(e)
+
+
+async def sign_file_with_autograph(context, from_, fmt, to=None):
+    """Signs a file with autograph and writes the result to arg `to` or `from_` if `to` is None.
+
+    Args:
+        context (Context): the signing context
+        from_ (str): the source file to sign
+        fmt (str): the format to sign with
+        to (str, optional): the target path to sign to. If None, overwrite
+            `from_`. Defaults to None.
+
+    Raises:
+        Requests.RequestException: on failure
+
+    Returns:
+        str: the path to the signed file
+
+    """
+    if not utils.is_autograph_signing_format(fmt):
+        raise SigningScriptError(
+            "No signing servers found supporting signing format {}".format(fmt)
+        )
+    cert_type = task.task_cert_type(context)
+    servers = get_suitable_signing_servers(context.signing_servers, cert_type, [fmt])
+    if not servers:
+        raise SigningScriptError(
+            "No signing servers found with cert type {}".format(cert_type)
+        )
+    s = servers[0]
+    to = to or from_
+
+    with open(from_, 'rb') as fin:
+        input_bytes = fin.read()
+
+    # build and run the signature request
+    sign_req = [{"input": base64.b64encode(input_bytes)}]
+    log.debug("using the default autograph keyid for %s", s.user)
+
+    url = "%s/sign/file" % s.server
+
+    async def make_sign_req():
+        auth = HawkAuth(id=s.user, key=s.password)
+        with requests.Session() as session:
+            r = session.post(url, json=sign_req, auth=auth)
+            log.info("Autograph response: {}".format(r.text))
```

Comment:
Nit: I would put this log as debug, and strip some parts of the response => when I run `test_autograph.sh` the test spew too much data to stdout.